### PR TITLE
Feature/41046

### PR DESCRIPTION
**Pull request by UrvashiChaudhary**
1. Create a Generic Table Component using Material Table Components, that will accept the followings props:
id: string, field name used to distinguish objects.
columns: array of objects, consisting of columns list that needs to be rendered
data: array of objects, data that needs to be rendered
Refer to “table-props.png”
2. Every object of *columns* props must have followings fields:
field: name of the field to be rendered.
label: optional, if provided then use label for the headers otherwise use the “field”
align: must be any among “left”, “right”, and “center”, and will be used to align the data of that particular fields.

### CODE REVIEW

File: training/src/components/Table/TableComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  table: {
+    minWidth: 650,
+  },
+  header: {
+    color: 'grey',
+  },
+
+});
+
+function TableComponent(props) {
```
Comment (adilsuccessive) [inline training/src/components/Table/TableComponent.jsx]: no need to write as TableComponent just Table is enough also change the file name.
Make this as arrow function or class component.

File: training/src/components/Table/TableComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+              </TableCell>
+              <TableCell>{email}</TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+}
+TableComponent.propTypes = {
```
Comment (adilsuccessive) [inline training/src/components/Table/TableComponent.jsx]: id is missing in prop.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]
label is not optional as per the task
label: optional, if provided then use label for the headers otherwise use the “field”
also table is not generic to work with different columns.
Table component must be generic component and will work with any number of columns.

File: training/src/components/Table/TableComponent.jsx (hunk 1 above)
Comment (UrvashiChaudhary) [reply training/src/components/Table/TableComponent.jsx]: done

File: training/src/components/Table/TableComponent.jsx (hunk 2 above)
Comment (UrvashiChaudhary) [reply training/src/components/Table/TableComponent.jsx]: done

**Comment by UrvashiChaudhary** [comment]
> label is not optional as per the task
> label: optional, if provided then use label for the headers otherwise use the “field”
> also table is not generic to work with different columns.
> Table component must be generic component and will work with any number of columns.

resolved

**Review by adilsuccessive — APPROVED** [review]

**Pull request merged**